decoder.go

```diff
@@ -247,8 +253,15 @@ func (d *decoder) unmarshalMap(size uint, offset uint, result reflect.Value) (ui
 
 func (d *decoder) unmarshalPointer(size uint, offset uint, result reflect.Value) (uint, error) {
 	pointer, newOffset := d.decodePointer(size, offset)
-	_, err := d.decode(pointer, result)
-	return newOffset, err
+
+	switch result.Kind() {
+	default:
+		_, err := d.decode(pointer, result)
+		return newOffset, err
+	case reflect.Uintptr:
+		result.Set(reflect.ValueOf(uintptr(pointer)))
+		return newOffset, nil
+	}
```

Comment:
This is de-referencing the pointer at `offset` and storing the result, whereas the case in `decode` is storing the referenced pointer _prior_ to decode.